### PR TITLE
Marketing analytics

### PR DESCRIPTION
**Pull request by motechFR**
Final todo: Send track page from public page in a way that it doesn't create duplicates

Need to test bounty and proposals, to see if they send this.

Main issue is we don't use the same CharmEditor in public and internal mode

### CODE REVIEW

File: pages/api/profile/index.ts [hunk 1]
```diff
@@ -32,29 +36,64 @@ async function createUser (req: NextApiRequest, res: NextApiResponse<LoggedInUse
...
+
   if (!profile) {
-    return res.status(404).json({ error: 'No user found' });
+    return handleNoProfile();
   }
+
+  // Clean up the anonymous id if the user has a profile
+  req.session.anonymousUserId = undefined;
+
+  await req.session.save();
```
Comment (mattcasey) [inline pages/api/profile/index.ts:95]: can you avoid this call if anonymouseUserId was not set? this call is called all the time, we should avoid extra async ops that are unecesssary

File: marketing-site/analytics.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<!--
```
Comment (mattcasey) [inline marketing-site/analytics.html:1]: please add a README to this folder explaining what the folder is for

File: hooks/useUserAcquisition.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      });
+    }
+
+  }
+
+  function setCampaign () {
+    const currentCampaign = router.query.utm_campaign;
+
+    if (currentCampaign) {
+      setCookie('appCampaign' as SignupCookieType, currentCampaign, {
```
Comment (mattcasey) [inline hooks/useUserAcquisition.ts:46]: please use setCookie/getCOokie from our utils in lib/utilities/browser - no need to add a new npm dependency

File: lib/metrics/mixpanel/utils-signup.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
+
```
Comment (mattcasey) [inline lib/metrics/mixpanel/utils-signup.ts:1]: all files should use camelCase

File: charmClient/apis/trackApi.ts [hunk 5]
```diff
@@ -1,10 +1,38 @@
...
+
+      return Promise.resolve({ success: 'ignored' });
+    }
+
+    this.lastEvent = {
+      payload: payloadAsString,
+      timestamp: now
+    };
+
+    return http.POST<{ success: 'ok' | 'ignored' }>(`/api/events?event=${event}`, payload);
```
Comment (mattcasey) [inline charmClient/apis/trackApi.ts:35]: the backend never returns 'ignored'?

File: pages/api/profile/index.ts [hunk 6]
```diff
@@ -32,29 +36,64 @@ async function createUser (req: NextApiRequest, res: NextApiResponse<LoggedInUse
...
   }
 
+  // Null out the anonmyous user id after successful login
+  req.session.anonymousUserId = undefined;
   req.session.user = { id: user.id };
   await updateGuildRolesForUser(user.wallets.map(w => w.address), user.spaceRoles);
   await req.session.save();
 
+  signupCookieNames.forEach(cookie => {
+    deleteCookie(cookie, {
```
Comment (mattcasey) [inline pages/api/profile/index.ts:57]: we have a pattern for handling cookies already on the backend
```
import Cookies from 'cookies';
....
const cookies = new Cookies(req, res);
...
cookies.set(cookie, null);
````

File: hooks/useUserAcquisition.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        maxAge: maxCookieAge,
+        path: '/'
+      });
+    }
+  }
+
+  function setLandingPage () {
+    const currentPage = window.location.href.split('?')[0];
+
+    if (currentPage) {
```
Comment (mattcasey) [inline hooks/useUserAcquisition.ts:32]: when would currentPage be undefined? seems impossible

File: pages/index.tsx [hunk 8]
```diff
@@ -26,6 +27,13 @@ export default function LoginPage () {
   const { spaces, isLoaded: isSpacesLoaded } = useSpaces();
   const discordCookie = getCookie(AUTH_CODE_COOKIE);
 
+  const { refreshSignupData } = useUserAcquisition();
+
+  useEffect(() => {
```
Comment (mattcasey) [inline pages/index.tsx:32]: this function needs to run on ever page, not just if users hit the domain w/no path (pages/index.tsx). I believe you just have to put this in app.tsx instead

**Review by mattcasey — CHANGES_REQUESTED** [review]
see comments

File: marketing-site/analytics.html (hunk 2 above)
Comment (motechFR) [reply marketing-site/analytics.html:1]: yep, done

File: hooks/useUserAcquisition.ts (hunk 3 above)
Comment (motechFR) [reply hooks/useUserAcquisition.ts:49]: Gotcha, will know for future!

File: charmClient/apis/trackApi.ts (hunk 5 above)
Comment (motechFR) [reply charmClient/apis/trackApi.ts:35]: Yep! Fair point

File: pages/api/profile/index.ts (hunk 6 above)
Comment (motechFR) [reply pages/api/profile/index.ts:57]: Gotcha, will know in future

File: hooks/useUserAcquisition.ts (hunk 7 above)
Comment (motechFR) [reply hooks/useUserAcquisition.ts]: I was going along with patter for other setters here, but you're right. Removed this :)

File: pages/index.tsx (hunk 8 above)
Comment (motechFR) [reply pages/index.tsx]: Done!

**Review by mattcasey — APPROVED** [review]

**Pull request merged**